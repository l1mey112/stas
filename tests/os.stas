include <std.stas>
include <salloc.stas>

fn test_normalise_path {
	auto old_brk 1
		salloc.now pop old_brk

	'../hello/../gello' normalise_path over strlen putuln puts endl

	old_brk salloc.rewind

	'//.///./' normalise_path over strlen putuln puts endl

	old_brk salloc.rewind

	'eer/../ree/../../../file.stas' normalise_path over strlen putuln puts endl

	old_brk salloc.rewind
}

fn test_join_path {
	auto old_brk 1
		salloc.now pop old_brk

	'hello///' 'executeable.exec' join_path_cstr over strlen putuln puts endl

	'hello' 'plebbit' join_path_cstr over strlen putuln puts endl

	old_brk salloc.rewind

	'hello/top/s/' 'top/plebbit' join_path_cstr over strlen putuln puts endl

	old_brk salloc.rewind
}

fn test_errno {
	EPERM          errno? bool.to_str puts endl
	EIO            errno? bool.to_str puts endl
	EFAULT  negate errno? bool.to_str puts endl
	EBADMSG negate errno? bool.to_str puts endl
	0              errno? bool.to_str puts endl
}

fn test_file_dir {
	'/usr/home/l-m/git/stas/lib/posix/makedefinitions.vsh' file_dir puts endl
	'/usr/home/l-m/git/compiler/tests/comptime.out' file_dir puts endl
	'' file_dir puts endl
	'eeeeeeeeeeee' file_dir puts endl
	'/eee' file_dir puts endl
}

fn main {
	test_normalise_path
	test_join_path
	test_errno
	test_file_dir

	"/hello"       is_abs_path bool.to_str puts endl
	"hello/heeler" is_abs_path bool.to_str puts endl
}