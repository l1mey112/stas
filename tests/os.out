8
../gello
1
/
15
../../file.stas
22
hello/executeable.exec
13
hello/plebbit
23
hello/top/s/top/plebbit
[1m[31mEPERM: Operation not permitted
[39m[22mfalse
[1m[31mEIO: Input/output error
[39m[22mfalse
[1m[31mEFAULT: Bad address
[39m[22mfalse
[1m[31mEBADMSG: Bad message
[39m[22mfalse
true
/usr/home/l-m/git/stas/lib/posix
/usr/home/l-m/git/compiler/tests
.
.
/
true
false
